paymentsheet/src/test/java/com/stripe/android/paymentelement/embedded/DefaultEmbeddedSheetLauncherTest.kt

```diff
@@ -68,97 +75,100 @@ internal class DefaultEmbeddedSheetLauncherTest {
-        assertThat(manageActivityLauncher.unregisterTurbine.awaitItem()).isEqualTo(Unit)
+        it.lifecycleOwner.handleLifecycleEvent(Lifecycle.Event.ON_DESTROY)
+        val formUnregisterCall = awaitUnregisterCall()
+        val manageUnregisterCall = awaitUnregisterCall()
+
+        assertThat(formUnregisterCall).isNotNull()
+        assertThat(manageUnregisterCall).isNotNull()
     }
 
     private fun testScenario(
-        block: suspend Scenario.() -> Unit,
+        activityResultCallerBlock: suspend DummyActivityResultCaller.Scenario.(launcherScenario: Scenario) -> Unit
     ) = runTest {
-        MockitoAnnotations.openMocks(this)
-        val activityResultCaller = mock<ActivityResultCaller>()
         val lifecycleOwner = TestLifecycleOwner()
-        val formActivityLauncher = FakeEmbeddedActivityLauncher(FormContract)
-        val manageActivityLauncher = FakeEmbeddedActivityLauncher(ManageContract)
         val savedStateHandle = SavedStateHandle()
         val selectionHolder = EmbeddedSelectionHolder(savedStateHandle)
         val customerStateHolder = CustomerStateHolder(savedStateHandle, selectionHolder.selection)
 
-        @Suppress("UNCHECKED_CAST")
-        val formContractCallbackCaptor: ArgumentCaptor<ActivityResultCallback<FormResult>> = ArgumentCaptor
-            .forClass(ActivityResultCallback::class.java) as ArgumentCaptor<ActivityResultCallback<FormResult>>
-
-        whenever(
-            activityResultCaller.registerForActivityResult(
-                any<FormContract>(),
-                capture(formContractCallbackCaptor)
+        DummyActivityResultCaller.test {
+            val sheetLauncher = DefaultEmbeddedSheetLauncher(
+                activityResultCaller = activityResultCaller,
+                lifecycleOwner = lifecycleOwner,
+                selectionHolder = selectionHolder,
+                customerStateHolder = customerStateHolder
             )
-        ).thenReturn(formActivityLauncher)
-
-        @Suppress("UNCHECKED_CAST")
-        val manageContractCallbackCaptor: ArgumentCaptor<ActivityResultCallback<ManageResult>> = ArgumentCaptor
-            .forClass(ActivityResultCallback::class.java) as ArgumentCaptor<ActivityResultCallback<ManageResult>>
-
-        whenever(
-            activityResultCaller.registerForActivityResult(
-                any<ManageContract>(),
-                capture(manageContractCallbackCaptor)
+            val formRegisterCall = awaitRegisterCall()
+            val manageRegisterCall = awaitRegisterCall()
+            val formLauncher = awaitNextRegisteredLauncher()
+            val manageLauncher = awaitNextRegisteredLauncher()
+
+            assertThat(formRegisterCall).isNotNull()
+            assertThat(manageRegisterCall).isNotNull()
+
+            assertThat(formRegisterCall.contract).isInstanceOf<FormContract>()
+            assertThat(formRegisterCall.callback).isInstanceOf<ActivityResultCallback<*>>()
```

Comment:
I don't think we care about the callbacks here. We should validate those work separately.